### PR TITLE
Less hacky map generation!

### PR DESCRIPTION
**Pull request by NegativeScript**

### CODE REVIEW

File: furusiya/model/components/walkers/random_walker.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    def walk(self):
+        valid_options = [0, 1, 2, 3]
+        random.shuffle(valid_options)
+
+        while valid_options:
+            x, y = self.parent.x, self.parent.y
+            n = valid_options.pop()
+
+            if n == 0:
```
Comment (nightblade9) [inline furusiya/model/components/walkers/random_walker.py]: lol speaking of horrible code that I wrote *ahem*

File: furusiya/model/maps/generators/forest_generator.py [hunk 2]
```diff
@@ -0,0 +1,256 @@
+from attrdict import AttrDict
```
Comment (nightblade9) [inline furusiya/model/maps/generators/forest_generator.py]: Why do we need this here?

File: furusiya/util_functions.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import tdl
```
Comment (nightblade9) [inline furusiya/util_functions.py]: I would rename this class. `util`/`helper` things quickly become dumping grounds for random things.

I would call it `targetting_functions`

File: furusiya/util_functions.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+from main_interface import Game
+from model.components.fighter import Fighter
+from view.renderer import render_all
+
+
+def target_monster(max_range=None):
+    # returns a clicked monster inside FOV up to a range, or None if right-clicked
```
Comment (nightblade9) [inline furusiya/util_functions.py]: These horrible docs I wrote eventually need to be docstring'd.

File: furusiya/view/renderer.py [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+        return False
+    elif not Game.area_map.tiles[x][y].is_walkable:
+        return False
+    elif Game.area_map.tiles[x][y].block_sight:
+        return False
+    else:
+        return True
+
+
+def get_names_under_mouse():
```
Comment (nightblade9) [inline furusiya/view/renderer.py]: Should be moved to `targetting_functions` no?

File: furusiya/view/renderer.py [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+    # create a list with the names of all objects at the mouse's coordinates and in FOV
+    names = [obj.name for obj in get_objects_under_mouse()]
+
+    names = ', '.join(names)  # join the names, separated by commas
+    return names.capitalize()
+
+
+def get_objects_under_mouse():
```
Comment (nightblade9) [inline furusiya/view/renderer.py]: Ditto under `targetting_functions`

File: furusiya/view/renderer.py [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+    (x, y) = Game.mouse_coord
+
+    # create a list with the names of all objects at the mouse's coordinates and in FOV
+    stuff = [obj for obj in Game.area_map.entities
+             if obj.x == x and obj.y == y and (obj.x, obj.y) in Game.visible_tiles]
+
+    return stuff
+
+
+def closest_monster(max_range):
```
Comment (nightblade9) [inline furusiya/view/renderer.py]: Ditto here (targetting functions deal with mouse, or targetting by distance -- maybe we can also split it along those lines into two files)

File: furusiya/view/renderer.py [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+
+import colors
+import config
+from constants import FOV_ALGO, FOV_LIGHT_WALLS, MAP_HEIGHT, MAP_WIDTH, color_dark_wall, color_dark_ground, \
+    color_light_wall, color_light_ground, MSG_X, PANEL_Y, SCREEN_WIDTH, PANEL_HEIGHT
+from main_interface import Game
+from model.components.fighter import Fighter
+
+
+def render_all():
```
Comment (nightblade9) [inline furusiya/view/renderer.py]: I seriously applaud your honestly heroic efforts at extracting a renderer out of this spaghetti.

File: furusiya/model/maps/generators/forest_generator.py (hunk 2 above)
Comment (NegativeScript) [reply furusiya/model/maps/generators/forest_generator.py]: Using it as a dummy object in _random_walk

File: furusiya/view/renderer.py (hunk 7 above)
Comment (NegativeScript) [reply furusiya/view/renderer.py]: Moving them would be easier if the code wasn't spaghetti.
It's hard not to fall into cyclic imports. They will be moved when the code is clean enough.

**Pull request merged**